redbot/cogs/streams/streamtypes.py

```diff
@@ -118,6 +131,9 @@ def __repr__(self):
 
 
 class Stream:
+
+    token_name: ClassVar[str]
```

Comment:
Since this attribute can be `None` you'll want to type-hint it as a `ClassVar[Optional[str]]`. You can import `Optional` from the typing module too. Also should probably initialize it to `None` here as in the suggestion.
```suggestion
    token_name: ClassVar[Optional[str]] = None
```